### ISSUE TITLE
Change Request: provide means for unsetting or changing the implicitly included global rules for **/*.js, **/*.cjs, and **/*.mjs

### ISSUE THREAD

**Issue by bladeSk** [eslint/eslint#19838]
### ESLint version

v9.28.0

### What problem do you want to solve?

Eslint always creates hidden rules for `**/*.js`, `**/*.cjs`, and `**/*.mjs`, so even without specifying anything, these files are linted. There are generated files like `storybook-static`, `dist` and any number of helper scripts in the project that shouldn't be linted.

I can disable this behavior using `globalIgnores(['**/*.js', '**/*.cjs', '**/*.mjs']),`, but this prevents me from linting `**/src/*.js`. Is there a recommended solution for this?

### What do you think is the correct solution?

Removing the hidden behavior would likely be a major breaking change so perhaps providing a utility to disable it would be a nice enough workaround:

```javascript
import { defineConfig, globalIgnores, noImplicitDefaults } from 'eslint/config'

defineConfig([
  noImplicitDefaults,
  ...
])
```

Or maybe `defineEmptyConfig()` as a counterpart to `defineConfig()`. Or any other way of disabling the default imports.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Can you explain more about what it is you're trying to do and the problem you're running into? It's not clear from your description what's happening.

**Comment by bladeSk** [comment]
I don't want `**/*.js` to be linted, only `src/**/*.js`. Ideally, I'd like the linter to lint _only_ what I specify and not add some hidden global rules.

To clarify further, I'm using `eslint.config.mjs` and running the CLI version of eslint via `npx eslint`.

**Comment by nzakas** [comment]
Okay, let me just clarify because you're using terminology in a way I'm unfamiliar with.

Rules are the things you enable in the `rules` section of your config, and you're in complete control of those. By "hidden global rules" I think you mean default glob patterns?

If you only want to match files in `src`, you can run ESLint like this:

```shell
npx eslint src
```

You can also do this to be ultra-specific:

```shell
npx eslint src/**/*.js
```

Passing the directory or a glob pattern to the CLI ensures only the files you want to lint are linted.
---

I'm assuming you're doing one of these:

```shell
npx eslint

# or

npx eslint .
```

In that case, ESLint is going to find all the files that match anything in your `eslint.config.js`, plus any default glob patterns. If you don't want the default, you'll have to ignore them and then unignore anything you do want to lint, such:

```js
import { defineConfig, globalIgnores } from 'eslint/config'

export default defineConfig([
  globalIgnores(['**/*.js', '**/*.cjs', '**/*.mjs', '!**/src/**/*.js'])
  ...
])
```

This is discussed in the docs:
https://eslint.org/docs/latest/use/configure/ignore#unignoring-files-and-directories

**Comment by bladeSk** [comment]
Okay, both of these solutions work great for what I wanted to achieve, so thank you for the hints.

I was debugging the config files with [Config Inspector](https://github.com/eslint/config-inspector), which doesn't allow specifying a glob pattern, so it always displayed the implicit rules for `eslint/defaults/files` and `eslint/defaults/files-cjs`, as I described above.

The confusion stems from the fact that I set up an `eslint.config.mjs` file with a single configuration object with a `files` pattern and expected `npx lint` to lint just the files specified there.

So while I'm happy with the solution, I find this behavior unintuitive. I'm not sure how this could be fixed without breaking anything, other than mentioning this more explicitly in the docs, so feel free to close the issue.

**Comment by nzakas** [comment]
Glad you got it working. As with using any new tool, it takes time to understand how to use it in the way that you want. ESLint does a lot of different things so that it's useful to the widest number of people and, unfortunately, that means not everything is instantly intuitive.
